data/pools/transactionPool.go

```diff
@@ -816,8 +816,24 @@ func (pool *TransactionPool) AssembleBlock(round basics.Round, deadline time.Tim
 						}
 					}
 					stats.TotalLength += uint64(encodedLen)
+					stats.StateProofNextRound = uint64(assembled.Block().StateProofTracking[protocol.StateProofBasic].StateProofNextRound)
 					if txib.Txn.Type == protocol.StateProofTx {
-						stats.HasStateProofNextRound = uint64(assembled.Block().StateProofTracking[protocol.StateProofBasic].StateProofNextRound)
+						lastSPRound := txib.Txn.StateProofTxnFields.StateProofIntervalLatestRound
```

Comment:
nits: should we create a function that returns the state proof `StateProofStats` if necessary? that would help to avoid the nested `if`s